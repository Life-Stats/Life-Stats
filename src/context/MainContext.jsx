import { createContext, useContext, useState } from 'react';

const MainContext = createContext();

const MainProvider = ({ children }) => {
  const [name, setName] = useState('');

  return (
    <MainContext.Provider value={{ name, setName }}>
      {children}
    </MainContext.Provider>
  );
};

const useMain = () => {
  const context = useContext(MainContext);

  if (context === undefined) {
    throw new Error('useMain must be used within a MainContext Provider');
  }
  return context;
};

export { MainProvider, useMain };