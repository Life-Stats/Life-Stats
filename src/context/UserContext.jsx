import { createContext, useContext, useMemo, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  // using the lazy initializer method would be good here
  const [user, setUser] = useState(() => {
    return currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  }
);

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

// Dont need to export the context since we use it in the useUser() hook
export { UserProvider, useUser };