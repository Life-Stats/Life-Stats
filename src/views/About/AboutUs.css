img {
  height: 175px;
  width: 175px;
}

.section {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  margin: 50px 150px 20px 150px;
}

.section img {
  border-radius: 3%;
  border: 4px rgba(254, 109, 115) solid;
  box-shadow: black 2px 2px 6px 2px ;
}

div.left, div.right {
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin: 50px 0 40px 0;
  display: flex;
  align-items: center;
  font-size: 17px;
}

div.right:last-child {
  margin-bottom: 300px;
}

.name {
  font-weight: bolder;
  font-size: 2em;
}

.text {
  padding: 0 15px 0 15px;
}

.right {
display: flex;
align-self: flex-end;
}

.left {
display: flex;
align-self: flex-start;
}

.aboutHeader {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  margin-top: 40px;
  border: 2px rgba(254, 109, 115) solid;
  width: 345px;
  box-shadow: black 2px 2px 6px 2px;
}