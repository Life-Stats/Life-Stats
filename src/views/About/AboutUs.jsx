import React from 'react'

export default function AboutUs() {
    return (
        <div>
            <h1>About Us view</h1>
        </div>
    )
}
