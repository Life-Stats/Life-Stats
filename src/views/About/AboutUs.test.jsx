import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import AboutUs from '../About/AboutUs';

it('should render AboutUs view', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <AboutUs />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
