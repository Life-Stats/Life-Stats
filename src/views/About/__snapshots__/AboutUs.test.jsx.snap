// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AboutUs view 1`] = `
<div>
  <div>
    <div
      class="aboutHeader"
    >
      <h1>
        Meet The Team
      </h1>
    </div>
    <section
      class="section"
    >
      <div
        class="left"
      >
        <p
          class="name"
        >
          Simon Kaine
        </p>
        <img
          src="/path/image"
        />
        <p
          class="text"
        >
          Hi! I am a Full Stack Software Engineer currently living in Milwaukie, Oregon. Born on May 31st 1988 when the number one song in the US that week was 'One More Try' by George Michael.
        </p>
        <div
          class="linkList"
        >
          <a
            href="https://www.linkedin.com/in/simonbishopkaine/"
          >
            <img
              class="linkedIn"
              src="/path/image"
            />
          </a>
          <a
            href="https://github.com/simonkaine"
          >
            <img
              class="gitHub"
              src="/path/image"
            />
          </a>
        </div>
      </div>
      <div
        class="right"
      >
        <p
          class="name"
        >
          Aemilius Morgan
        </p>
        <img
          src="/path/image"
        />
        <p
          class="text"
        >
          Hello! I am a software engineer located in Portland OR. an interesting fact about me is that I am left handed, but I can use both fairly well.
        </p>
        <div
          class="linkList"
        >
          <a
            href="https://www.linkedin.com/in/aemilius-morgan/"
          >
            <img
              class="linkedIn"
              src="/path/image"
            />
          </a>
          <a
            href="https://github.com/AemiliusM"
          >
            <img
              class="gitHub"
              src="/path/image"
            />
          </a>
        </div>
      </div>
      <div
        class="left"
      >
        <p
          class="name"
        >
          Peter Montano
        </p>
        <img
          src="/path/image"
        />
        <p
          class="text"
        >
          Full Stack Software Developer currently living in Portland, Oregon. Born on August 12th 1998, the year of the tiger.
        </p>
        <div
          class="linkList"
        >
          <a
            href="https://www.linkedin.com/in/petermontano/"
          >
            <img
              class="linkedIn"
              src="/path/image"
            />
          </a>
          <a
            href="https://github.com/Montano-Pete"
          >
            <img
              class="gitHub"
              src="/path/image"
            />
          </a>
        </div>
      </div>
      <div
        class="right"
      >
        <p
          class="name"
        >
          Dylan Greathouse
        </p>
        <img
          src="/path/image"
        />
        <p
          class="text"
        >
          Developing a passion for front and back end web development. Love any opportunity to design and create. Always looking for a chance to learn and be a better version of myself than I was the day before. Strong interest in game design and cooking.
        </p>
        <div
          class="linkList"
        >
          <a
            href="https://www.linkedin.com/in/dylan-greathouse/"
          >
            <img
              class="linkedIn"
              src="/path/image"
            />
          </a>
          <a
            href="https://github.com/Dylan-Greathouse"
          >
            <img
              class="gitHub"
              src="/path/image"
            />
          </a>
        </div>
      </div>
    </section>
  </div>
</div>
`;
